web/src/server/api/routers/models.ts

```diff
@@ -1,69 +1,248 @@
+import { v4 as uuidv4 } from "uuid";
 import { z } from "zod";
 
-import { ModelUsageUnit } from "@langfuse/shared";
+import { auditLog } from "@/src/features/audit-logs/auditLog";
+import { isValidPostgresRegex } from "@/src/features/models/server/isValidPostgresRegex";
+import {
+  GetModelResultSchema,
+  UpsertModelSchema,
+} from "@/src/features/models/validation";
 import { throwIfNoProjectAccess } from "@/src/features/rbac/utils/checkProjectAccess";
 import {
   createTRPCRouter,
   protectedProjectProcedure,
 } from "@/src/server/api/trpc";
-import { paginationZod } from "@langfuse/shared";
+import { ModelUsageUnit, paginationZod } from "@langfuse/shared";
 import { TRPCError } from "@trpc/server";
-import { auditLog } from "@/src/features/audit-logs/auditLog";
-import { isValidPostgresRegex } from "@/src/features/models/server/isValidPostgresRegex";
 
 const ModelAllOptions = z.object({
   projectId: z.string(),
   ...paginationZod,
 });
 
 export const modelRouter = createTRPCRouter({
-  all: protectedProjectProcedure
+  getById: protectedProjectProcedure
+    .input(z.object({ projectId: z.string(), modelId: z.string() }))
+    .query(async ({ input, ctx }) => {
+      const modelQueryResult = await ctx.prisma.$queryRaw`          
+          SELECT 
+            m.id,
+            m.project_id as "projectId",
+            m.model_name as "modelName",
+            m.match_pattern as "matchPattern",
+            m.tokenizer_config as "tokenizerConfig",
+            m.tokenizer_id as "tokenizerId",
+            COALESCE(
+              (
+                SELECT
+                  JSONB_OBJECT_AGG(usage_type, price)
+                FROM
+                  prices
+                WHERE
+                  model_id = m.id
+              ),
+              '{}'::jsonb
+            ) AS prices
+          FROM
+            models m
+          WHERE
+            m.id = ${input.modelId}
+            AND (
+              project_id IS NULL
+              OR project_id = ${input.projectId}
+            );
+      `;
+
+      const model = z.array(GetModelResultSchema).parse(modelQueryResult)[0];
+
+      if (!model || (model.projectId && model.projectId !== input.projectId)) {
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message: "Model not found",
+        });
+      }
+
+      return model;
+    }),
+
+  getAll: protectedProjectProcedure
```

Comment:
do we have model read roles? If yes, we should add this here.

Reply:
.. and above.

Reply:
we only have `models:CUD` for now, read is open to all users